### DISCUSSION TITLE
Make UI Compact for mobile devices

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#2361] [UI Related]
We know that the UI is broken on iOS:

![5D0563EC-4EB4-42A9-9992-5158076428D2](https://user-images.githubusercontent.com/20726856/152954510-5c88901c-e2d1-41db-b3b9-82b1031a1008.jpeg)

![1F45F4FA-5271-476E-98B3-25B31B63FB9E](https://user-images.githubusercontent.com/20726856/152954624-e6b302ce-02b7-484a-9da6-0740ae1d6d06.png)
Without QML we probably can’t get a native UI; however I think we can improve the UI:
1. Make reverb buttons say L/R instead of left/right.
2. Make connect button smaller in compact skin
3. Settings/Chat/Mute myself can be renamed to something short in compact skin: Set./Ch./Mte M.
4. Ping/Jitter/Delay could also be made brief (P,J,D)
5. Input could be named "In"
Edit: switching to Chinese might give a feeling how it could look. 

![6E99E255-548A-428E-813F-BA60BC21C698](https://user-images.githubusercontent.com/20726856/152958349-6a643a1b-057d-40c6-965d-6227937416d4.png)
![3BE63FD9-136D-4F42-AA82-A2AD285838BC](https://user-images.githubusercontent.com/20726856/152958357-af65b5ba-40b4-46aa-9211-231b0f82e252.png)
![A91688BD-1B63-4F77-9515-BA7C09414570](https://user-images.githubusercontent.com/20726856/152958365-350a3b51-2b4d-4dc6-aa45-b961deb8fbad.png)
![2EC6EE2F-D661-48CE-8B37-3424B71E6702](https://user-images.githubusercontent.com/20726856/152958367-81811200-3f59-4dbd-89d2-710da15181b9.png)

**Comment by dcorson-ticino-com** [comment]
I still think one should not use a screw driver when one needs a hammer.

**Reply by ann0see** [reply]
Yes, but I doubt moving to QML is something we're doing soon

**Reply by dcorson-ticino-com** [reply]
Sorry for my ignorance, what is QML and how does it make a screen bigger ?

**Reply by ann0see** [reply]
It doesn’t make the screen bigger but seems to be the modern way of describing the UI. I think it’s somewhat similar to HTML: https://doc.qt.io/qt-6/qmlapplications.html

**Comment by ann0see** [comment]
Otherwise, it may make sense to move the reverb slider to advanced settings - similar to how we have the input balance and boost in Advanced settings.

**Reply by gilgongo** [reply]
> Otherwise, it may make sense to move the reverb slider to advanced settings - similar to how we have the input balance and boost in Advanced settings.

I agree that might be a good thing to do if it makes mobile easier. At first sight, the position of the reverb slider looks like it could be an input gain, and I wonder how often people adjust reverb once it's set (if they set it at all). By that logic, it could be classed as a "setting"?

**Reply by gilgongo** [reply]
This reminds me I've also thought having a logo graphic is a bit unconventional for a desktop app. But I suppose removing it wouldn't do much for the mobile view.

**Comment by pljones** [comment]
Before we start changing _anything_ in the UI, I'd get all platforms onto Qt 6.7.  Then take stock of what's broken.  From that point, the aim should be to keep the UI "reactive" or where that's not possible "mobile first".  The UI isn't sufficiently complex to require the desktop version to be radically different, in my view.

(So yes, why use a screwdriver when you need a hammer, indeed.)

**Comment by dcorson-ticino-com** [comment]
I guess that if people want to use devices with more latency as they only have wireless connectivity they should be able to.
Personally I don't understand the reasoning, mobile devices should be the last thing on the list..

**Reply by ann0see** [reply]
Cellular 5G latency isn't that bad. 
Also mobile is more convenient and people use it for listening in.
